api/v1beta2/bucket_types.go

```diff
@@ -49,6 +49,7 @@ const (
 
 // BucketSpec specifies the required configuration to produce an Artifact for
 // an object storage bucket.
+// +kubebuilder:validation:XValidation:rule="self.provider != 'aws' || !has(self.sts) || self.sts.provider == 'aws'", message="'aws' is the only supported STS provider for the 'aws' Bucket provider"
```

Comment:
I was able to test this CEL validation locally with k8s 1.30 (and adding `ldap` as an accepted value for `spec.sts.provider`, in-cluster only, just for the test since this will be introduced in a separate PR).

Reply:
I was actually worried about this, thanks for finding a way to write automatic tests for this!

Reply:
Fixed, thanks! Please let me know if these are enough.

Reply:
By the way, I just renamed `TestBucketReconciler_CELValidation_STS` to `TestBucketReconciler_APIServerValidation_STS` and added the test case below:

```
		{
			name:           "invalid endpoint",
			bucketProvider: "aws",
			stsConfig: &bucketv1.BucketSTSSpec{
				Provider: "aws",
				Endpoint: "test",
			},
			err: "spec.sts.endpoint in body should match '^(http|https)://.*$'",
		},
```

Please let me know if that is okay.

Reply:
Fixed, thanks!